src/contracts/core/DelegationManager.sol

```diff
@@ -797,13 +807,36 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         }
     }
 
-    function _initializeWithdrawalDelayBlocks(uint256 _withdrawalDelayBlocks) internal {
+    /**
+     * @notice Sets the withdrawal delay blocks for each strategy in `_strategiesToSetDelayBlocks` to `_withdrawalDelayBlocks`.
+     * only gets called once when initializing the contract.
+     */
+    function _initializeStrategyWithdrawalDelayBlocks(
+        IStrategy[] calldata _strategiesToSetDelayBlocks,
+        uint256[] calldata _withdrawalDelayBlocks
+    ) internal {
         require(
-            _withdrawalDelayBlocks <= MAX_WITHDRAWAL_DELAY_BLOCKS,
-            "DelegationManager._initializeWithdrawalDelayBlocks: _withdrawalDelayBlocks cannot be > MAX_WITHDRAWAL_DELAY_BLOCKS"
+            _strategiesToSetDelayBlocks.length == _withdrawalDelayBlocks.length,
+            "DelegationManager._initializeStrategyWithdrawalDelayBlocks: input length mismatch"
         );
-        emit WithdrawalDelayBlocksSet(withdrawalDelayBlocks, _withdrawalDelayBlocks);
-        withdrawalDelayBlocks = _withdrawalDelayBlocks;
+        uint256 numStrats = _strategiesToSetDelayBlocks.length;
+        for (uint256 i = 0; i < numStrats; ++i) {
+            IStrategy strategy = _strategiesToSetDelayBlocks[i];
+            uint256 prevStrategyWithdrawalDelayBlocks = strategyWithdrawalDelayBlocks[strategy];
+            uint256 newStrategyWithdrawalDelayBlocks = _withdrawalDelayBlocks[i];
+            require(
+                newStrategyWithdrawalDelayBlocks <= MAX_WITHDRAWAL_DELAY_BLOCKS,
```

Comment:
Do we need to change this value? I think the proposal wants a different max delay.

Reply:
yes. need to lengthen this to I think 4 weeks.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/392/commits/323fa4e387941179468d541a4c9d9470c2382bd1#diff-4d12f7b9016c76f80ea4d90dc4e49dbddda5177ec6d7c8cd4055b2f3f857a20aR49